test/example/type-builder.cpp

```diff
@@ -0,0 +1,137 @@
+#include <cassert>
+#include <iostream>
+
+#include "wasm-type.h"
+
+using namespace wasm;
+
+// Construct Signature, Struct, and Array heap types using undefined types.
+void test_builder() {
+  std::cout << ";; Test TypeBuilder\n";
+
+  // (type $sig (func (param (ref $struct)) (result (ref $array) i32)))
+  // (type $struct (struct (field (ref null $array) (mut ref extern))))
+  // (type $array (array (mut ref null extern)))
```

Comment:
```suggestion
  // (type $struct (struct (field (ref null $array) (mut externref))))
  // (type $array (array (mut externref)))
```
Nit: For simplicity (and also to match the output below)

Reply:
Good catch, thanks! I'll replace the second one (`ref null extern`), but the first one `ref extern` is not the same as `externref` because it is non-nullable.